### PR TITLE
intel_s1000: doc: Update doc with MCUbootloader support

### PR DESCRIPTION
**Pull request by rgundi**
The documentation for intel_s1000 is updated with the steps
to use it with MCUbootloader.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

### CODE REVIEW

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 1]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
 (simple shorting without the resistor will also do) for debugging to work.
 Those 2 pins are Pin5 HOST_RST_N_LT_R) and Pin21 (+V_HOST_3P3_1P8).
 
+MCUbootloader Support
+=====================
+
+Intel S1000 on Zephyr supports the open source, cross-RTOS MCUbootloader.
```
Comment (nashif) [inline boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: replace MCUbootloader  with MCUBoot Bootloader

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 1 above)
Comment (sathishkuttan) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Also, it's Zephyr on Intel S1000, not Intel S1000 on Zephyr.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 2]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+MCUbootloader in conjuntion with USB DFU (Device Firmware Upgrade)
+over the air can be used for Field upgradation of Intel S1000 Firmware.
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: USB DFU is not exactly over the "air", please remove "over the air". Also change "upgradation" to "upgrades".

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 3]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+`MCUboot GitHub repo`_ page. Check out `MCUboot with Zephyr`_ documentation
+page on the MCUboot website for more on MCUboot.
+
+First, you need to generate the MCUbootloader image for Intel S1000.
+Go to the mcubootloader repository on your local machine and execute the
+below commands.
+
+.. code-block:: console
+
+   cd boot/zephyr
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: This block is similar to the app build and should be able to use the `zephyr-app-commands` directive. See around line 136

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 4]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+   cmake -DBOARD=intel_s1000_crb ..
+   make
+
+zephyr/zephyr.bin is the output of the above steps and this needs to be passed
+through a utility 'create_spi_flash' which will pad this binary suitably so
+that this gets recognized by the S1000 BootROM. Below is the command.
+
+.. code-block:: console
+
+   ./create_spi_flash zephyr/zephyr.bin mcuboot.bin -sk
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: create_spi_flash utility is not available publicly. need to make it or a variant of it as part of the build system before updating the document.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 5]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+
+mcuboot.bin is the MCUbootloader which now needs to be flashed to address 0 in
+the SPI flash. Please ensure the flash is completely erased before loading the
+MCUbootloader.
+
+MCUbootloader requires 2 images viz. image0 (default image) and image1 (new
+image). The image0 should have DFU capability as this image gets loaded by
```
Comment (sathishkuttan) [inline boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: So, when the active image is switched to image1, will the DFU capability be lost? If the DFU capability is part of the image, then both image0 and image1 should have the capability.

**Review by sathishkuttan — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 2 above)
Comment (dbkinder) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Will this procedure (and upgrading the S1000 firmware) work on the board referenced at the top of this doc (https://software.intel.com/en-us/iot/speech-enabling-dev-kit)?  I suspect not, and we need to be more explicit at the top of this doc about the board and procedures being documented here.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 1 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Corrected.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 2 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: @dbkinder This procedure will work on the board referenced at the top of this doc.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 3 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: zephyr-app-commands are applicable only to those apps which are within the zephyr tree. mcubootloader is outside of it.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 4 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: ok. noted. I will keep this for the time being while i work on the utility

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 5 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: That's correct. updated the doc.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 6]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+1. Your application's :file:`.conf` file needs to enable the
+   :option:`CONFIG_BOOTLOADER_MCUBOOT` Kconfig option in order for Zephyr to
+   be built in an MCUboot-compatible manner. For example, if you want to build
+   tests/boards/intel_s1000_crb as MCUboot-compatible, you need to enable the
+   Kconfig option :option:`CONFIG_BOOTLOADER_MCUBOOT` in prj.conf.
+
+2. The application image thus built above must be signed. Use the
+   ``scripts/imagetool.py`` script from the `MCUboot GitHub repo`_
+   to sign the image. Below is an example (See the
+  `MCUboot with Zephyr`_ documentation for details.)
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 7]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
 (simple shorting without the resistor will also do) for debugging to work.
 Those 2 pins are Pin5 HOST_RST_N_LT_R) and Pin21 (+V_HOST_3P3_1P8).
 
+MCUbootloader Support
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 8]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+=====================
+
+Intel S1000 on Zephyr supports the open source, cross-RTOS MCUbootloader.
+MCUbootloader in conjuntion with USB DFU (Device Firmware Upgrade)
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 9]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+over the air can be used for Field upgradation of Intel S1000 Firmware.
+The MCUbootloader source code itself is hosted in the
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 10]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+
+mcuboot.bin is the MCUbootloader which now needs to be flashed to address 0 in
+the SPI flash. Please ensure the flash is completely erased before loading the
+MCUbootloader.
+
+MCUbootloader requires 2 images viz. image0 (default image) and image1 (new
+image). The image0 should have DFU capability as this image gets loaded by
+the MCUbootloader and is the active image during a Firmware upgrade. The flash
+offsets into which image0 and image1 need to be flashed can be found out by
+looking at boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts. Refer to
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 11]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+:ref:`flash_partitions` for details about partition layout.
+
+In order to build an MCUboot-compatible Intel S1000 Firmware image
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 12]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+(image0/image1), you need to do the following:
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 13]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+
+1. Your application's :file:`.conf` file needs to enable the
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Done.

**Comment by rgundi** [comment]
@dbkinder @sathishkuttan addressed your comments. Plz have a look.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst [hunk 14]
```diff
@@ -218,10 +218,98 @@ However, we need to short 2 pins on Host Connector J3 via a 3.3k resistor
...
+
+.. code-block:: console
+
+   cd boot/zephyr
+   mkdir build
+   cd build
+   cmake -DBOARD=intel_s1000_crb ..
+   make
+
+zephyr/zephyr.bin is the output of the above steps and this needs to be passed
```
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: yes. That's correct.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Good point. As of now, we have not exposed this utility. I'll try to make changes to invoke this utility implicitly. In that case, this line won't be required at all.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 3 above)
Comment (sathishkuttan) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: see `:app:` directive (vs `:zephyr-app:`) at https://docs.zephyrproject.org/latest/documentation/doc-guidelines.html#zephyr-app-commands-directive

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (dbkinder) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: OK, but it's still here in this version of the documentation being reviewed.

**Review by dbkinder — CHANGES_REQUESTED** [review]
Docs look better, but still need to resolve use of ./create_spi_flash

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 3 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: modified as suggested.

File: boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst (hunk 4 above)
Comment (rgundi) [reply boards/xtensa/intel_s1000_crb/doc/intel_s1000.rst]: Removed this now. https://github.com/zephyrproject-rtos/zephyr/pull/12662 addresses this.

**Comment by rgundi** [comment]
> Docs look better, but still need to resolve use of ./create_spi_flash

The reference to create_spi_flash is now removed. Please review again.

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

**Comment by rgundi** [comment]
@nashif .. please check if this can be merged. Shippable failures seem to be unrelated.

**Comment by nashif** [comment]
stale

**Pull request closed without merging**